sphere-models/src/it/java/io/sphere/sdk/products/queries/QueryAllTest.java

```diff
@@ -0,0 +1,72 @@
+package io.sphere.sdk.products.queries;
+
+import io.sphere.sdk.models.Identifiable;
+import io.sphere.sdk.products.Product;
+import io.sphere.sdk.queries.PagedQueryResult;
+import io.sphere.sdk.queries.PagedResult;
+import org.junit.Test;
+
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Optional;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CompletionStage;
+
+import static java.util.stream.Collectors.toList;
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class QueryAllTest extends QueryAllBase {
+    //yes, this is a small page size, but since this is tested,
+    //we create not million products for test execution speed reasons
+    //in production code it would higher
+    private static final int PAGE_SIZE = 5;
+
+    @Test
+    public void useIdPredicateInsteadOfOffset() throws Exception {
+        final ProductQuery seedQuery = ProductQuery.of()
```

Comment:
ProductQuery can be used for both "products" and "product-projections" endpoint?